static/js/subs.js

```diff
@@ -860,18 +860,30 @@ $(function () {
         selectText(this);
     });
 
-    $("#subscriptions_table").on("click", ".sub_unsub_button", function (e) {
-        e.preventDefault();
-        e.stopPropagation();
+    function sub_or_unsub (e) {
+      e.preventDefault();
+      e.stopPropagation();
+      var stream_name = $(e.target).data("name");
+
+      if (!stream_name) {
```

Comment:
You should add a comment here explaining the store around these two cases.

Actually, a cleaner approach would be to have `sub_or_unsub` just take the `stream_name` as an argument and have the two event handlers each do their own way of extracting the name, rather than doing both in the same function.